### PR TITLE
whitneys code review

### PR DESCRIPTION
**Pull request by whitneyann**
criticize me

### CODE REVIEW

File: app_base.htm [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  % endfor
+</ul>
+
+</%block>
+
+<%block name ="right">
+<h2 class="recent">Recently Viewed:</h2>
+<ul>
+  <%
+    if request.dmp_router_page == 'details':
```
Comment (aspruill) [inline app_base.htm:63]: I like how this takes into account that it shouldn't show the current product you are on if its in the details.

File: details.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    for p in request.last_five:
+        if product == p:
+            request.last_five.remove(p)
+
+    request.last_five.insert(0,product)
+
+    while len(request.last_five)>6:
+        request.last_five.pop()
+
+
```
Comment (aspruill) [inline details.py:22]: It's nice how concise this is. It might be helpful to include comments about how things function.

File: details.css [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+.desc{
+  width:35%;
+  float:right;
+}
+
+p{
+  float:right;
+}
+
+.menu{
```
Comment (mattkastner) [inline details.css]: wow, menu

File: history.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # idlist= []
```
Comment (aspruill) [inline history.py:8]: Before submitting this you should delete these comments, that way the file is more concise.  And like I said before, you should add comments to explain how things function.

File: app_base.htm (hunk 1 above)
Comment (mattkastner) [reply app_base.htm]: lol hehe

File: app_base.htm [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+<%block name="left">
+<ul class = 'menu_hover'>
+  <!-- ## <li><a class= "menu" href= '/catalog/index/'>All Products</a></li> -->
+  <li>
+  <a class = '${'menu' if name == 'All Products' else ''}' href="/catalog/index/0/">All Products</a>
+  </li>
+  <%! from catalog import models as cmod %>
+  % for Category in cmod.Category.objects.all():
+    <li><a class = '${'menu' if name == Category.name else ''}' href= '/catalog/index/${Category.id}/' >${Category.name}</a></li>
```
Comment (aspruill) [inline app_base.htm:52]: The href part in this is really good. It adds clarity to the code.

File: details.html [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+<div class='desc'>
+  <p>
+    ${product.description}
+    <br /> Jowl strip steak brisket pork beef pig. Kevin sirloin spare ribs, swine pork buffalo
+     pork chop cow alcatra short ribs andouille ribeye pork belly pastrami. Kevin chicken capicola jerky tenderloin. Beef cow pork loin bacon.
+  </p>
+  <br /><br />
+  <a href="/" class="btn btn-danger btn-lg" role="button">Buy</a></button>
+  ## %if product.quantity > 1:
```
Comment (aspruill) [inline details.html:34]: Your code would totally look cleaner without these funky comments

File: details.html [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    ## %if pic != main_pic:
+        <img src= "${pic}" class='smallpic' alt= '${product.name}'/>
+    ## %endif
+  %endfor
+</div>
+
+<div class='desc'>
+  <p>
+    ${product.description}
+    <br /> Jowl strip steak brisket pork beef pig. Kevin sirloin spare ribs, swine pork buffalo
```
Comment (aspruill) [inline details.html:29]: Did you just add a random comment to the end of the description? That potentially ruins the accuracy of the description.

File: details.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from catalog import models as cmod
+import math
+from django.http import HttpResponseRedirect
+
+@view_function
```
Comment (aspruill) [inline details.py:8]: Before submitting your code in the future, I think you should submit your entire file to me for review. Just to make sure it's okay.

File: app_base.htm [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        <li class = ${'active' if request.dmp_router_page == 'sections' else ''}><a href="/sections/">Sections</a></li>
+      </ul>
+
+      % if request.user.is_authenticated:
+        <ul class="nav navbar-nav navbar-right">
+          <li class="dropdown">
+          <a class="dropdown-toggle" data-toggle="dropdown" href="#">Welcome ${user.first_name}
+          <span class="caret"></span></a>
+            <ul class="dropdown-menu">
+                <li><a href="/account/logout/">Logout</a></li>
```
Comment (gihongChoi) [inline app_base.htm:26]: correct indentation might help with readability here

File: app_base.htm [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+            </ul>
+        </ul>
+      % else:
+        <ul class="nav navbar-nav navbar-right">
+          <li class="dropdown">
+          <a class="dropdown-toggle" data-toggle="dropdown" href="#">Account
+          <span class="caret"></span></a>
```
Comment (gihongChoi) [inline app_base.htm:33]: maybe put the closing a tag in a separate line?

File: details.html [hunk 11]
```diff
@@ -0,0 +1,48 @@
+## <%inherit file="/homepage/templates/base_ajax.htm"/>
```
Comment (gihongChoi) [inline details.html:1]: this comment seems unnecessary?

File: details.html [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+     pork chop cow alcatra short ribs andouille ribeye pork belly pastrami. Kevin chicken capicola jerky tenderloin. Beef cow pork loin bacon.
+  </p>
+  <br /><br />
+  <a href="/" class="btn btn-danger btn-lg" role="button">Buy</a></button>
+  ## %if product.quantity > 1:
+  ##   Quantity: <input type="number" name="quantity" min="1" max='1000'>
+  ## %endif
+  %if product.TITLE== "Bulk":
+    <select>
+      %for n in range(product.quantity):
```
Comment (gihongChoi) [inline details.html:39]: I think variable name range here can be a little more descriptive of what it holds.

File: details.py (hunk 8 above)
Comment (gihongChoi) [reply details.py:8]: Wow

File: history.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        # try:
+        #     idlist=request.session['lastfive'] #doesn't get only return one object?
+        # #convert product ids into objects- how do I do that?
+        # except:
+        #     "Error"
```
Comment (gihongChoi) [inline history.py:13]: remove these comments?
